### DISCUSSION TITLE
Social preview with icon and friendly text

### DISCUSSION THREAD

**Discussion by djjudas21** [navidrome/navidrome#2294] [Ideas]
I would like to request some kind of nice-looking preview when Navidrome links are pasted into social media. Obviously most people have their Navidrome instance authenticated so it won't be possible to show details about a specific album/artist/track, but it would be nice if:
- a preview could be generated with the Navidrome icon and maybe a friendly sentence like `Modern Music Server and Streamer compatible with Subsonic/Airsonic`
- ideally, the name of the Navidrome server and the tag line could be overridden for display purposes, e.g. I might want to replace `Navidrome` with `Jonathan's Music` and the tag line with `Here are my sweet choons`

**Here's how a link to Navidrome renders in Facebook**

![Screenshot from 2023-04-05 15-26-44](https://user-images.githubusercontent.com/604595/230113256-caeb07dc-2a19-496f-8c1e-649a0d5c11bf.png)

**Here's how a link to Navidrome renders in Slack**

![Screenshot from 2023-04-05 15-30-35](https://user-images.githubusercontent.com/604595/230113313-57ddc191-35e3-4615-802b-e1745288d45b.png)

**Comment by deluan** [comment]
Shared links (using the "Share" feature #2106) have a proper social preview, including coverart. Wouldn't it be better to use that instead of links to an authenticated page?

**Comment by djjudas21** [comment]
Thanks for the response @deluan :smile: yes the share feature works nicely but only allows access to one album, and only allows access to to stream but not download.

My use case is a bit weird - I'm using it as a public streaming service (against advice, I understand) for music from my church. I want public users to be able to browse the albums, play them, download them, etc. So for me, this URL is not actually authenticated. It's running in Kubernetes from a read-only file system, so the danger is minimal. But it means we can publish the short URL on service sheets, and also share links to specific albums or tracks on social media etc.

So yes, I'm aware that I'm abusing Navidrome, but it would still be cool if the social previews of the authenticated endpoint could show something nicer than just the version string.

This is the site: https://music.stmarysfishponds.org.uk/

And here's how I have configured it: https://jonathangazeley.com/2022/03/18/making-a-public-music-streaming-service-with-navidrome/

Side note: even if the Navidrome instance was authenticated in the normal way, I think social previews of the instance would still be useful, because it is still conceivable that you might want to make someone an account and send them a link.

**Reply by deluan** [reply]
Just to be clear, the Share feature allows you to share not only albums, but playlists and artists as well. And in the next version it will allow downloads: https://github.com/navidrome/navidrome/pull/2246

Anyway, I now understand your use case and, as you rightly mentioned, it is against advice. Having said that, I'll take a look and see if I can make a better social preview.

**Reply by djjudas21** [reply]
Thanks for the clarification - I played with the share feature a bit. I'm looking forward for downloadable shares, but for me I think a killer feature would be the ability to browse content within a share - which is why I'm using the main UI.

Thanks for looking at the social preview too :slightly_smiling_face:
